### ISSUE TITLE
Gather state-level healthcare facility bed counts for all possible states

### ISSUE THREAD

**Issue by lossyrob** [daveluo/covid19-healthsystemcapacity#14]
The goal of this issue is to generate a dataset for every healthcare facility in every US state possible that contains the number of licensed beds as well as licensed ICU beds. Licensed beds are different then the HCRIS data "staffed" beds in that the former is the number of beds the hospital is licensed to operate; the latter is the number of beds that were actively staffed (and is usually lower and a worse approximation for maximum capacity).

The data generated should be of a standard format for each state. Each state has a different format for the  information, so there should be a transformation for each state's data to put it into a common format. Part of this issue can be establishing the destination format, e.g. from the from the first state scraped.

What we really want to know is the # of licensed beds (how many beds that facility is legally allowed to staff, if staffing at 100%), along with the number of beds that are ICU beds (if available). In every state, this is reported to that state's health facility licensing authority. but only in some states are they easily accessible in open data portals, e.g.:
- NY
  - https://health.data.ny.gov/Health/Health-Facility-Map/875v-tpc8
  - https://health.data.ny.gov/Health/Adult-Care-Facility-Directory/wssx-idhx
  - https://health.data.ny.gov/Health/Health-Facility-Certification-Information/2g9y-7kqm
- CA
  - https://healthdata.gov/dataset/licensed-and-certified-healthcare-facility-bed-types-and-counts
- NJ
  - https://www.nj.gov/cgi-bin/dhss/healthfacilities/hospitaldisplay.pl?id=10402
MA:
  - https://www.mass.gov/service-details/find-information-about-licensed-or-certified-health-care-facilities

Even here, the formatting of the data on the portal, and in the spreadsheets, and even the distinction between bed types can be very different.

States may have ICU bed information so that an ICU bed count can be tallied; see https://gitter.im/covid19-healthsystemcapacity/community?at=5e6ef57d1f0d285eb280772c

Here's a good "in the wild" example of staffed vs licensed vs available beds as policed by a state authority: https://oshpd.ca.gov/ml/v1/resources/document?rs:path=/Data-And-Reports/Documents/Submit/Financial/Hospital/Hospital-Technical-Letters/hsptechltr30.pdf
on page 2

Here's another one from a state that's more visual and informative: https://esf8.dhh.la.gov/documentportal/Download/Public/Definitions%20of%20Bed%20Poll%20Terms.pdf
as you can see, this is not straightforward even to hospitals that have to report the data!

However, if we were able to go state-by-state and scrape the data together for every state that had the data open on the internet, we could go a long way towards improving the accuracy of our healthcare system capacity dataset.

See the Gitter conversation starting at https://gitter.im/covid19-healthsystemcapacity/community?at=5e6eec1ac8ddaf238c8991ad

**Comment by lossyrob** [comment]
These are the target statistics we're aiming to derive with this data:

- total licensed non-ICU beds & % of these staffed
- total licensed ICU beds & % of these staffed
- total number of vents & % of these available

Where:

- Licensed Beds: The maximum number of beds for which a hospital holds a license to operate. Many hospitals do not operate all of the beds for which they are licensed. The number of licensed beds is a better proxy for the number of beds available during a staffing surge.
- Staffed Beds: Beds that are licensed and physically available for which staff is on hand to attend to the patient who occupies the bed. Staffed beds include those that are occupied and those that are vacant. These numbers reported are for Business As Usual (BAU), and underestimate the maximum number of beds available.

States may use different terms then "staffed" or "licensed". Whatever the number that best represents these definitions should be used to convert the state-level data into something we can combine with other state data and HCRIS data.

@daveluo can you check my definitions here?

**Comment by lossyrob** [comment]
Primer for calculating occupancy rates: https://www.health.pa.gov/topics/HealthStatistics/Statistical-Resources/UnderstandingHealthStats/Documents/Occupancy_Rates_in_Health_Facilities.pdf

**Comment by daveluo** [comment]
Thanks! Here are the definitions of various bed counts issued by [Agency for Healthcare Research and Quality (ARHQ)](https://archive.ahrq.gov/research/havbed/definitions.htm). I **bolded** the 2 we care most about right now:

- **Licensed Beds**: The maximum number of beds for which a hospital holds a license to operate. Many hospitals do not operate all of the beds for which they are licensed.**
- Physically Available Beds: Beds that are licensed, physically set up, and available for use. These are beds regularly maintained in the hospital for the use of patients, which furnish accommodations with supporting services (such as food, laundry, and housekeeping). These beds may or may not be staffed but are physically available.
- **Staffed Beds**: Beds that are licensed and physically available for which staff is on hand to attend to the patient who occupies the bed. Staffed beds include those that are occupied and those that are vacant.
- Unstaffed Beds: Beds that are licensed and physically available and have no current staff on hand to attend to a patient who would occupy the bed.
- Occupied Beds: Beds that are licensed, physically available, staffed, and occupied by a patient.
- Vacant/Available Beds: Beds that are vacant and to which patients can be transported immediately. These must include supporting space, equipment, medical material, ancillary and support services, and staff to operate under normal circumstances. These beds are licensed, physically available, and have staff on hand to attend to the patient who occupies the bed.

Great figure showing the relationship of these definitions:
![beds relationship diagram](https://archive.ahrq.gov/research/havbed/beds.gif)

Re:
>States may use different terms then "staffed" or "licensed". Whatever the number that best represents these definitions should be used to convert the state-level data into something we can combine with other state data and HCRIS data.

We can make the reasonable assumption that all facilities in all states are (or should be) reporting according to these standardized defns. Note that there will be some stray definition misunderstandings or data entry errors that should be somewhat obvious as extreme outliers (like the 2 billion total beds reported by 1 facility). We can put in some ballpark sanity checks on the numbers we collect and drop or manually correct the ones that are way out of the expected distribution or impossible (like having more reported ICU beds than total beds).

**Comment by daveluo** [comment]
Here is a potentially great source for licensed bed counts for all US health facilities, with long/lat coordinates as well: https://hifld-geoplatform.opendata.arcgis.com/datasets/hospitals

h/t to this project: https://github.com/rbracco/covidcompare, https://forums.fast.ai/t/help-with-algorithm-for-covid19-relative-risk/65328

Upon initial review of the data, it looks like they pulled the Licensed Bed counts from each state's health facility licensing authority (based on SOURCE column), same as the state-level sources we've listed for CA, WA, NY, etc

The geocoding seems to be validated by inspecting imagery (based on VAL_METHOD column).

I did a quick pivot table of the data here: https://docs.google.com/spreadsheets/d/1NFqXFOJ9TlXwR0xaKdJrI3wsQSOlvFz3wZpxE_GIuvA/edit?usp=sharing

The total listed bed count for Open Critical Access + Acute Care facilities to be 833K which sounds about right.

We could merge this with our hospital dataset spatially by doing a buffered point join. Then we'll have both listed total bed count and staffed bed counts. Unfortunately this dataset doesn't give ICU beds alone but based on the HCRIS data's staffed ICU bed:total bed counts, we can estimate the # of listed ICU beds.
